From e9d298d94ec86760cc74b8be45739d061f583b46 Mon Sep 17 00:00:00 2001
From: Bharat Bhushan <Bharat.Bhushan@freescale.com>
Date: Thu, 7 Jan 2016 17:06:33 +0530
Subject: [PATCH 02/15] Revert "dpio-driver: Fix qman-portal interrupt masking
 in poll mode"

This reverts commit 99ab4f9b8d4555b6a7f554d1f2e0e11fdc24a50c.

Signed-off-by: Bharat Bhushan <Bharat.Bhushan@freescale.com>

Conflicts:
	drivers/staging/fsl-mc/bus/dpio/dpio-drv.c
---
 drivers/staging/fsl-mc/bus/dpio/dpio-drv.c | 29 ++++++++++++-----------------
 1 file changed, 12 insertions(+), 17 deletions(-)

diff --git a/drivers/staging/fsl-mc/bus/dpio/dpio-drv.c b/drivers/staging/fsl-mc/bus/dpio/dpio-drv.c
index 80add27..5389ee8 100644
--- a/drivers/staging/fsl-mc/bus/dpio/dpio-drv.c
+++ b/drivers/staging/fsl-mc/bus/dpio/dpio-drv.c
@@ -216,30 +216,25 @@ dpaa2_dpio_probe(struct fsl_mc_device *ls_dev)
 		resource_size(&ls_dev->regions[0]));
 	desc.regs_cinh = ioremap(ls_dev->regions[1].start,
 		resource_size(&ls_dev->regions[1]));
+	priv->io = dpaa_io_create(&desc);
+	if (!priv->io) {
+		dev_err(dev, "DPIO setup failed\n");
+		goto err_dpaa_io_create;
+	}
 
 	err = fsl_mc_allocate_irqs(ls_dev);
 	if (err) {
 		dev_err(dev, "DPIO fsl_mc_allocate_irqs failed\n");
-		desc.has_irq = 0;
-	} else {
-		irq_allocated = true;
-
-		snprintf(priv->irq_name, MAX_DPIO_IRQ_NAME, "FSL DPIO %d",
-			 desc.dpio_id);
-
-		err = register_dpio_irq_handlers(ls_dev, desc.cpu);
-		if (err)
-			desc.has_irq = 0;
+		goto poll_mode;
 	}
+	irq_allocated = true;
 
-	priv->io = dpaa2_io_create(&desc);
-	if (!priv->io) {
-		dev_err(dev, "DPIO setup failed\n");
-		goto err_dpaa2_io_create;
-	}
+	snprintf(priv->irq_name, MAX_DPIO_IRQ_NAME, "FSL DPIO %d",
+			desc.dpio_id);
 
-	/* If no irq then go to poll mode */
-	if (desc.has_irq == 0) {
+	err = register_dpio_irq_handlers(ls_dev, desc.cpu);
+poll_mode:
+	if (err) {
 		dev_info(dev, "Using polling mode for DPIO %d\n",
 			 desc.dpio_id);
 		/* goto err_register_dpio_irq; */
-- 
1.8.3.2

